#Parameter file for the Mantis MM-UAV

#Hex
#motor:
#  num: 6
#  arm_len: 0.275 #m
#  #thrust_max: 8.5 #11.6110736 # N
#  kv: 660 #Rated to 960, but real is ~660 under load
#  rpm_thrust_curve:
#    m: 0.0011323 #0.0008318056
#    c: -2.9235
#  drag_max: 1.2 # N

#Quad
motor:
  num: 4
  arm_len: 0.225 #m
  kv: 1240
  rpm_thrust_curve:
    m: 0.0005554685
    c: -0.139859613
  drag_max: 1.2 # N


airframe: "quad_x4"

pwm:
  min: 1000
  max: 2000

body:
  num: 3
  b0:
    mass:
      m: 1.44 #2.38 #Base + props
      Ixx: 0.02850657
      Iyy: 0.02821766
      Izz: 0.05416778
      com: 0.0
    link:
      name: "mount"
      d: -0.10 # Update this
      t: 0.0
      r: 0.0
      a: 1.570796326794897
      type: "static"
  b1:
    mass:
      m: 0.1
      Ixx: 0.00005
      Iyy: 0.00068
      Izz: 0.00067
      com: 0.15
    link:
      name: "joint_shoulder"
      d: 0.0
      t: -1.570796326794897
      r: 0.24
      a: 0.0
      type: "twist_z"
      beta: 0.8 # Maybe lower the filter a bit, or maybe set to 1 to remove it?
  b2:
    mass:
      m: 0.06
      Ixx: 0.00004
      Iyy: 0.00032
      Izz: 0.00032
      com: 0.11
    link:
      name: "joint_elbow"
      d: 0.0
      t: 0.0
      r: 0.21
      a: 0.0
      type: "twist_z"
      beta: 0.8 # Maybe lower the filter a bit, or maybe set to 1 to remove it?
